src/Showtime.jsx

```diff
@@ -70,5 +70,24 @@ export default function Showtime({
         }
     }, [status, previousStatus, onHidden, onShowing]);
 
-    return isMounted ? children(ref) : null;
+    // If child is a function like (ref) => <Child ref={ref} />
+    if (typeof children === "function") {
+        return isMounted ? children(showtimeRef) : null;
+    }
+
+    // Otherwise it is a component <Child />, to which ref must be attached
+    const child = Children.only(children);
+
+    return isMounted
+        ? cloneElement(child, {
+              ref(node) {
+                  showtimeRef.current = node;
+
+                  // Attach existing refs on child, if any
+                  if (child.ref?.hasOwnProperty("current")) {
+                      child.ref.current = node;
+                  }
```

Comment:
I think you also need to handle the case of an inbound _callback_ ref…

```
else if (typeof child.ref === "function") {
    child.ref(node);
}
```

Reply:
Added in 9a2ab8a

Reply:
Since there's now 4 references to `child.ref`, might be nice to deconstruct child first: `const { ref } = child;`

Reply:
Deconstructed in 9a2ab8a